source/constructs/api/data_source/service.py

```diff
@@ -2581,29 +2671,356 @@ def query_connection_detail(account: JDBCInstanceSourceBase):
     conn['ConnectionProperties']['JDBC_CONNECTION_SCHEMA'] = source.jdbc_connection_schema
     return conn
 
+def __gen_resources_name(jdbc):
+    database_type = convert_provider_id_2_database_type(jdbc.account_provider_id)
+    glue_connection_name = f"{const.SOLUTION_NAME}-{database_type}-{jdbc.instance_id}"
+    glue_database_name = glue_connection_name
+    crawler_name = f"{const.SOLUTION_NAME}-{database_type}-{jdbc.instance_id}"
+    return glue_connection_name, glue_database_name, crawler_name
 
 def __get_excludes_file_exts():
     extensions = list(set([ext for extensions_list in const.UNSTRUCTURED_FILES.values() for ext in extensions_list]))
     return ["*.{" + ",".join(extensions) + "}"]
 
+def __get_glue_client(account, region):
+    iam_role_name = crud.get_iam_role(account)
+    assumed_role = sts.assume_role(
+        RoleArn=f"{iam_role_name}",
+        RoleSessionName="glue-connection"
+    )
+    credentials = assumed_role['Credentials']
+    glue = boto3.client('glue',
+                        aws_access_key_id=credentials['AccessKeyId'],
+                        aws_secret_access_key=credentials['SecretAccessKey'],
+                        aws_session_token=credentials['SessionToken'],
+                        region_name=region
+                        )
+    return glue
+
+# def list_jdbc_databases(source: JdbcSource) -> list[str]:
+#     url_arr = source.connection_url.split(":")
+#     if len(url_arr) != 4:
+#         raise BizException(MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_code(), MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_msg())
+#     if url_arr[1] != "mysql":
+#         raise BizException(MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_code(), MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_msg())
+#     host = url_arr[2][2:]
+#     port = int(url_arr[3].split("/")[0])
+#     user = source.username
+#     password = source.password
+#     if source.secret_id:
+#         secrets_client = boto3.client('secretsmanager')
+#         secret_response = secrets_client.get_secret_value(SecretId=source.secret_id)
+#         secrets = json.loads(secret_response['SecretString'])
+#         user = secrets['username']
+#         password = secrets['password']
+#     mysql_database = jdbc_database.MySQLDatabase(host, port, user, password)
+#     databases = mysql_database.list_databases()
+#     logger.info(databases)
+#     return databases
+
+def batch_create(file: UploadFile = File(...)):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 65 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks on lines 2735-2773, lines 2775-2793 into separate functions.